docker/swift-t/Dockerfile

```diff
@@ -10,18 +10,27 @@ FROM 136a88385611
 ENV SWIFT_ROOT=/opt/swift-t
 
 # Install dependencies
-RUN yum -y install mpich-devel # OpenMPI is fine too
+# RUN yum -y install mpich-devel
+RUN yum -y install openmpi-devel
 RUN yum -y install swig
 RUN yum -y install zsh
-RUN wget https://prdownloads.sourceforge.net/tcl/tcl8.6.11-src.tar.gz
+RUN mkdir -pv /tmp/build-tcl
+WORKDIR /tmp/build-tcl
+RUN wget --no-verbose https://prdownloads.sourceforge.net/tcl/tcl8.6.11-src.tar.gz
 RUN tar xfz tcl8.6.11-src.tar.gz
+# Fix Tcl RTLD setting:
+RUN cd tcl8.6.11/unix \
+  && sed -i 90s@^@//@    tclLoadDl.c \
+  && sed -i 92,+2s@^@//@ tclLoadDl.c
 RUN cd tcl8.6.11/unix \
   && ./configure --prefix=/opt/tcl-8.6.11 \
-  && make -j binaries libraries \
+  && make -j 4 binaries libraries \
```

Comment:
I have seen cases where 4-way parallel make could lead to a hang when the CI pipeline is run on a completely resource-constraint cloud instance. This may be an area where we can trade off performance with robustness...